### PR TITLE
Add github actions docker images cache

### PR DESCRIPTION
**Pull request by usernamedt**
Add cache for docker images to speed up the retry times of integration tests

### CODE REVIEW

**Comment by reshke** [comment]
LGTM? by the way, if you are making a WIP PR, think about doing it in draft

File: .github/workflows/dockertests.yml [hunk 1]
```diff
@@ -22,19 +22,34 @@ jobs:
     name: Build images
     runs-on: ubuntu-16.04
     steps:
+      # if this is a retry job, there should cached image artifact available
```
Comment (reshke) [inline .github/workflows/dockertests.yml:25]: :cool:

**Review by reshke — APPROVED** [review]

**Comment by usernamedt** [comment]
> LGTM? by the way, if you are making a WIP PR, think about doing it in draft

I plan to update this PR tomorrow and improve some things

**Comment by usernamedt** [comment]
I've completely replaced build artifacts with caching, it should make retries even faster

**Pull request merged**